the-collab-lab/tcl-7-smart-shopping-list
# Duplicate Item Error Alert

## Description

In this task we ensured that the user could not input a duplicate item into the shopping list. When the user enters an item more than once
 
`Ex:` Typing in grapes, presses add and then re-types grapes and tries to add to the list, there is an error message that is prompted. 

The error message that is displayed is 'item is already in list'.  In the `addItem` function, we set up a database query that will recognize if the item has already been added. If the item has been added an error message is displayed letting the user know that the item is already in the shopping list. If the item does not exist, it will be saved to the database and added to the shopping list view page. 

## Related Issue

"closes" issue #7 

## Acceptance Criteria

- Show an error message if the user tries to submit a new item that has the exact same name as an existing item

- Show an error message if the user tries to submit a new item that has the same name as an existing item, where capitalization has been normalized and punctuation has been removed
- The user’s original input should be what gets saved in the database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|   ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|   ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

`N/A`


### After




## Testing Steps / QA Criteria

- From your terminal, pull down this branch with `git pull origin dt-tc-security-alert` and check that branch out with `git checkout dt-tc-security-alert`

- Then `npm install` to install the newly added dependencies locally and `npm start` to launch the app.

- In your browser navigate to the localhost server that is indicated in your terminal after `npm start`.

`NOTE:`
We are working on the 'Duplicate item error alert button/message box and the Normalize capitalization
